### ISSUE TITLE
Framer Motion v4 breaks Chakra 

### ISSUE THREAD

**Issue by brennanfife** [chakra-ui/chakra-ui#3618]
# 🐛 Bug report

## 💥 Steps to reproduce

1. Install a fresh instance of Chakra, including all of its required dependencies
2. Start the app
3. Notice it crash

## 💻 Link to reproduction

CodeSandbox reproduction: https://codesandbox.io/s/chakra-framer-motion-v4-bug-4i6rq

## 📝 Additional information

Looking at the Framer Motion [changelog](https://github.com/framer/motion/blob/main/CHANGELOG.md), it appears to be related to the deprecation of `motion.custom()` in favor of `motion()`, which I believe is being tracked in #3459 .

**Comment by philipguedes** [comment]
Same here, just installed Chakra in a new Nextjs project and got the same error

```
TypeError: _framerMotion.motion.custom is not a function
    at Object.<anonymous> (/node_modules/@chakra-ui/checkbox/dist/cjs/checkbox-icon.js:20:38)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
```

Downgraded `framer-motion` to `v3.10.6` and it solved the problem for me

**Comment by vickOnRails** [comment]
@philipguedes's answer helped me

- Uninstalled `framer-motion`
- Cleaned up `node_modules`
- Installed `framer-motion` v `3.10.6`

Everything worked flawlessly.

**Comment by intercaetera** [comment]
Perhaps before the actual code is updated, it would be good to push a quick fix to the docs. The installation command in [the docs](https://chakra-ui.com/docs/getting-started#installation) installs the wrong version of `framer-motion`.

**Comment by hgaskin** [comment]
I am new to this and experimenting with Chakra UI, here is the Error i received: 

(i ran npx create-react-app my-app --template @chakra-ui)

`TypeError: framer_motion__WEBPACK_IMPORTED_MODULE_1__.motion.custom is not a function
`
any help is resolving this would be greatly appreciated/

**Comment by with-heart** [comment]
Installations are breaking because our `peerDependencies` range for our packages specifies `"framer-motion": ">=3.0.0"`, which is too wide because we clearly don't actually support v4 yet. I've opened #3619 to specify `major` versions for the peer dependencies (which includes `framer-motion`) to at least avoid this in the future.

Currently unsure what we're going to do to handle the `peerDependencies` range being too open.

@hgaskin The instructions from @vickOnRails should resolve your issue.

**Comment by bhatvikrant** [comment]
# TLDR; 

`yarn add framer-motion@3.10.6`

instead of 

`yarn add framer-motion  //(this installs the v4 which is the latest version as of now.)`

This will work until official support for v4 of framer-motion is released in chakra-ui.

**Comment by luigiminardim** [comment]
I'm using `"framer-motion": "^3.10.6"` and I received this warning while developing with chakra-ui:
```
motion.custom() is deprecated. Use motion() instead.
```

This might be the solution to the compatibility problem.

**Comment by abda2050** [comment]
> I'm using `"framer-motion": "^3.10.6"` and I received this warning while developing with chakra-ui:
> 
> ```
> motion.custom() is deprecated. Use motion() instead.
> ```
> 
> This code change might be the solution to the compatibility problem since is this method that breaks when I used the `"framer-motion": "4.0.0"`

Thanks, @luigiminardim  this works for me right now the new version of Framer motion breaks my next app, I think the issue is that Chakra uses motion.custom() somewhere where with the new version of framer motion it's been changed into motion()

**Comment by mattgperry** [comment]
Sorry you're all running into this! IMO a good fix would be for the Chakra docs to pin `framer-motion` to a major, for instance `yarn add framer-motion@^3`.

For Chakra to upgrade to `4` the fix is indeed to change `motion.custom()` to `motion()`.

**Comment by lourou** [comment]
> Sorry you're all running into this! IMO a good fix would be for the Chakra docs to pin `framer-motion` to a major, for instance `yarn add framer-motion@^3`.
> 
> For Chakra to upgrade to `4` the fix is indeed to change `motion.custom()` to `motion()`.

This works fine! Thanks a lot

**Comment by with-heart** [comment]
The main thing I've been worried about in trying to resolve this situation for us is maintaining our commitment to semantic versioning. For a bit there, I was worrying that we had dug ourselves into a hole that we couldn't get out of, because forcing users to `4.0.0` would need to be considered a breaking change since `framer-motion` is a `peerDependency`.

However, I think that worry was incorrect. Our `peerDependency` range already supports `framer-motion@4.0.0` (because `>=3.0.0`), it's just that our codebase doesn't support it. Since we're not removing `^3.0.0` from this range, it's not a breaking change on our part, just a `minor` change that adds support for `framer-motion@^4.0.0` to the codebase.

Here's what the changes that will be part of this `minor` bump will consist of:

### Conditional checks to support both v3 and v4 APIs

We can determine whether we're in v3 or v4 based on the existence of `motion.custom`:

```tsx
const v3 = 'custom' in motion
const v4 = !('custom' in motion)
```

We can use this to add conditionals throughout the codebase in order to support both version:

- Conditionally call `motion` or `motion.custom` based on whether `motion.custom` exists:

```tsx
const MotionSvg =
  'custom' in motion ? (motion as any).custom(chakra.svg) : motion(chakra.svg)
```

- Bump `devDependencies` versions of `framer-motion` from `^3.1.1` to `^4.0.0`. Without this, we can't track changes to the v4 API and update our internals as necessary.

### Future-proofing

- Reduce `peerDependency` range of `framer-motion` to `3.x || 4.x`, so we can avoid this situation in the future with `framer-motion@5` 😬 
- Update installation instructions to install `framer-motion@^4` so that fresh Chakra installs use the current latest version. Since we'll eventually drop 3.0 support to take full advantage of the new bundle size reduction stuff in 4.0, this will make it easier for new installs to upgrade when that inevitably happens.

**Comment by Michellebuchiokonicha** [comment]
This just experienced the framer issue error again. What canbe done to resolve this issue permanently?
ps: I have used chakra-ui a number of times. I don't know why it came up this time. Does it have someting to do with the new React18?

**Comment by Michellebuchiokonicha** [comment]
I tried the solutions listed above to no avail. I need help

**Comment by bibatjed** [comment]
I have the same issue with react18. framer-motion  ^6 version.

**Comment by SDrinkwater** [comment]
It appears that this issue is back. I got it when starting a fresh project with the CRA typescript template.

**Comment by trompx** [comment]
I have the following error `TypeError: tslib.__spreadArray is not a function` with the following versions installed:
```json
"@chakra-ui/react": "^2.2.1",
"framer-motion": "^6.3.3",
react": "^18.2.0",
```
